src/main/scala/fi/oph/koski/valpas/ValpasOppijaSearchService.scala

```diff
@@ -105,7 +105,10 @@ class ValpasOppijaSearchService(application: KoskiApplication) extends Logging {
           // Henkilö, jonka tiedot löytyvät, mutta jolla maksuttomuus on päättynyt esim. toiselta asteelta
           // valmistumiseen, ei ole enää maksuttomuuden piirissä:
           case Some(_) => ValpasEiLainTaiMaksuttomuudenPiirissäHenkilöhakuResult()
-          case _ => ValpasEiLöytynytHenkilöhakuResult()
+          case None => henkilöRepository.findByOid(henkilö.oid) match {
```

Comment:
Pohdiskelen, että onko henkilö.oid tässä kohtaa varmasti henkilön master oid, eikä se, minkä käyttäjä on hakukenttään mahdollisesti syöttänyt, mikä ei välttämättä ole master. Kotikuntatieto varmaan pitäisi tutkia nimenomaan master oidilta.

Olisi varmaan selkeämpää, jos tietotyyppien ja niiden kenttien nimeämisestä jo kävisi ilmi, sisältääkö palautettu tai käyttäjältä tullut henkilö oid master oidin vai henkilön minkä vaan oidin... Mutta siinä refaktoroinnissa olisi kyllä varmaan aika iso jumppa.

Reply:
Miten ihmeessä sulla on silmää aina tällaisille yksityiskohdille? 😁

Tähän nimittäin näyttäisi tulevan slave-oid. Huomasin myös että tässä tehdään mahdollisesti turha toinen kysely onriin ja sen saa ehkä optimoitua pois.